# Titulo importante con un #
## titulo menos importante con 2# 
### titulo menos importante con 3 #
cualquier texto sin #

# Tipos de variables en python

In [1]:
3 #numeros

3

In [2]:
x=3 #asigacion de un valor de una variable 

In [3]:
print(x)

3


In [4]:
type(x)

int

In [5]:
x=3.0

In [6]:
x

3.0

In [7]:
type(x) #el tipo de variable cambio debdo al contenido que se le asigno

float

In [8]:
variable=5

In [9]:
variable

5

In [10]:
variable+5


10

In [11]:
variable = variable+5
print(variable) #print devuelve en pantalla el valor dentro del los ()

10


In [12]:
variable = x
print(type(x))

<class 'float'>


In [13]:
x = "string" #las comillas nos indican que es una cadena de caracteres

In [14]:
type(x)

str

In [15]:
x

'string'

In [16]:
variable = "x"

In [17]:
variable

'x'

In [18]:
print(variable)

x


In [19]:
print(x)

string


In [20]:
variable=10

In [22]:
type(variable)

int

In [23]:
variable=8**6
variable

262144

In [24]:
8**6

262144

----

# Operaciones con variables

Con las variables que ya estudiamos, podemos definir operaciones.  

**Suma** 

In [1]:
3 + 7

10

De igual manera, podemos realizar la suma con dos variables definidas, es decir

In [1]:
x = 3
y = 7

In [2]:
x + y

10

Podemos sumar:

* `float` con `float` 
* `float` con `int`
* `str` con `str`

`float` con `float`

In [11]:
x = 1.2314
y = 3.1415
x + y 

4.3729000000000005

Nos regresará un `float`

In [12]:
type(x+y)

float

`float` con `int`

In [13]:
x = 1.2314
y = 3
x + y 

4.2314

Nos regresará un `float`

In [14]:
type(x+y)

float

`str` con `str`

In [15]:
x = 'Buenas '
y = 'Tardes'
x + y

'Buenas Tardes'

Nos regresará un `str`

In [16]:
type(x+y)

str

---
**Sustracción**

In [1]:
3 - 7

10

In [1]:
x = 10 
y = 5.5
x-y

4.5

Podemos restar:

* `float` con `float` 
* `float` con `int`

---
**Multiplicación**

In [2]:
3 * 7

21

In [3]:
x = 9 
y = 5.3
x*y

47.699999999999996

Aquí podemos observar algo **interesante**. Si hacemos la multiplicación a "mano" $9 \times 5.3 = 47.7$ ¿Qué pasa aquí? A esto se le conoce como _**Error de redondeo**_, este es ocasionado por Python a la hora de almacenar los números decimales en binario. Existen bibliotecas que nos evitan este problema, pero en general, todos los lenguajes de programación tienen este problema. 

Podemos mutiplicar:

* `float` con `float` 
* `float` con `int`

---
**División**

In [4]:
10/4

2.5

In [5]:
x = 20 
y = 13.2
x/y

1.5151515151515151

Podemos dividir:

* `float` con `float` 
* `float` con `int`

Como podemos esperar, dividir entre cero nos genera error:

In [6]:
5/0

ZeroDivisionError: division by zero

Python respeta la jerarquía de operaciones, en caso de combinar operaciones también se puede hacer uso de los parentesis.

---
**Potencia**

In [9]:
2.5**4 #Elevar 2.5 a la 4

39.0625

In [10]:
9**0.5 #Raíz cuadrada

3.0

In [12]:
(-1)**(0.5)#Existen también numeros complejos en Python

(6.123233995736766e-17+1j)

In [14]:
10**-4 

0.0001

In [18]:
x = 3
y = 3.2
x**y

33.63473536961897

Podemos potenciar:

* `float` con `float` 
* `float` con `int`

---
**Módulo**

Es el también conocido como resto de la división. 

Tenemos que _dividendo = divisor*cociente + resto_

Cuando la división es exacta, el módulo es cero. 

In [26]:
4 % 2 #Si dividimos 4 entre 2, da 2, no sobra y por tanto el módulo es 0.

0

In [25]:
7 % 3 #Si dividimos 7 entre 3, da 2 y nos sobra 1, por lo que el módulo es 1.

1

In [27]:
x = 10 
y = 5.5
x % y

4.5

La herramienta del módulo nos puede ayudar, por ejemplo, a encontrar números pares o multiplos de otros números. 